src/Loading/sceneLoader.ts

```diff
@@ -417,7 +417,13 @@ export class SceneLoader {
         SceneLoader.OnPluginActivatedObservable.notifyObservers(plugin);
 
         if (directLoad) {
-            onSuccess(plugin, plugin.directLoad ? plugin.directLoad(scene, directLoad) : directLoad);
+            if (plugin.directLoad) {
+                plugin.directLoad(scene, directLoad).then((data) => {

```

Comment:
You should also handle errors and call the onError callback.

Reply:
Done.